### PR TITLE
Improve reliability of ParallelSender and ShadowSender

### PR DESCRIPTION
**Pull request by tnleeuw**
- Fix ParallelSender and ShadowSender when messages are not repeatable
- Fix ShadowSender when not waiting for all senders to finish, and pipeline finishes before all senders have finished.
-

### CODE REVIEW

File: core/src/test/java/nl/nn/adapterframework/senders/ShadowSenderTest.java [hunk 1]
```diff
@@ -211,6 +212,60 @@ public void testResultSenderResultWith3ShadowSenders() throws Exception {
...
+
+		// Act
+		inputMessage.closeOnCloseOf(session, sender);
+		String result = sender.sendMessageOrThrow(inputMessage, session).asString();
+		session.close();
+
+		// Assert
+		assertEquals(ORIGINAL_SENDER_RESULT, result);
+
+		Thread.sleep(3000); // wait for results to be collected in the background
```
Comment (jkosternl) [inline core/src/test/java/nl/nn/adapterframework/senders/ShadowSenderTest.java:236]: Is it easy to convert this to use `await()` method, so this unittest perhaps runs faster when possible?

**Review by jkosternl — APPROVED** [review]
Just wondering why it worked before 😉

**Comment by tnleeuw** [comment]
> Just wondering why it worked before 😉

I don't think it ever did, but some things were just never used in that particular way. 😃

File: core/src/test/java/nl/nn/adapterframework/senders/ShadowSenderTest.java (hunk 1 above)
Comment (tnleeuw) [reply core/src/test/java/nl/nn/adapterframework/senders/ShadowSenderTest.java:236]: Not sure. Will look at it.

File: core/src/main/java/nl/nn/adapterframework/senders/ShadowSender.java [hunk 2]
```diff
@@ -137,30 +152,37 @@ public SenderResult sendMessage(Message message, PipeLineSession session) throws
 		try {
 			primaryGuard.waitForAllResources();
 		} catch (InterruptedException e) {
-			throw new SenderException(getLogPrefix()+"was interupted", e);
+			throw new SenderException(getLogPrefix() + "was interrupted", e);
```
[the remark this answers is not in the document]
Comment (tnleeuw) [reply core/src/main/java/nl/nn/adapterframework/senders/ShadowSender.java:155]: I was unsure. I think we probably should.

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**